src/t_string.c

```diff
@@ -35,8 +35,8 @@
  *----------------------------------------------------------------------------*/
 
 static int checkStringLength(client *c, long long size) {
-    if (size > 512*1024*1024) {
-        addReplyError(c,"string exceeds maximum allowed size (512MB)");
```

Comment:
should RM_StringTruncate need this change?
```
int RM_StringTruncate(RedisModuleKey *key, size_t newlen) {
    if (!(key->mode & REDISMODULE_WRITE)) return REDISMODULE_ERR;
    if (key->value && key->value->type != OBJ_STRING) return REDISMODULE_ERR;
    if (newlen > 512*1024*1024) return REDISMODULE_ERR;
```

Reply:
i don't think so... personally, i think this limit should not be part of that API at all. but since it's there and documented, i don't think we can change it (or even bind it to the config), that would be a breaking change for the API.